### ISSUE TITLE
Missing WITHHASH option in GEOxxx commands

### ISSUE THREAD

**Issue by zhongxinghong** [redis-rs/redis-rs#1347]
see https://redis.io/docs/latest/commands/georadius/

**Comment by nihohit** [comment]
hi, PRs are welcome :)

**Comment by zhongxinghong** [comment]
I try to add hash field to RadiusSearchResult, but I found out there might be some ambiguity when parsing dist & hash. The dist with f64 type could be parsed from Value::Int and the hash with u64 could alse be parsed from Value::Int too. Should we limit the type of dist to Value::XXXString and the type of hash to Value::Int? :(

**Comment by nihohit** [comment]
> Should we limit the type of dist to Value::XXXString and the type of hash to Value::Int?

Would that help? AFAIS the hash could still be parsed as string if no dist was provided. 

I don't think that there's a way to disambiguate a response to a request that has WITHHASH without coords or dist, from a response to a request that has WITHDIST without coords or hash. According to the docs, WITHHASH is a niche feature, so IMO the simplest solution is to require that WITHHASH be set only on requests that also set WITHDIST, so there'll be no ambiguity - the first value is always dist, the second is always hash.

**Comment by nihohit** [comment]
> so IMO the simplest solution is to require that WITHHASH be set only on requests that also set WITHDIST, so there'll be no ambiguity - the first value is always dist, the second is always hash.

What about this solution?

**Comment by somechris** [comment]
While the commit message of #2193 contained "Fixes #1347", landing it did not close this bug.

It think this bug was addressed in the 2.x branch and can get closed.

**Comment by nihohit** [comment]
@somechris I think the PR you linked to is the wrong PR - it's still open, and doesn't say it closes this issue.

**Comment by somechris** [comment]
> [@somechris](https://github.com/somechris) I think the PR you linked to is the wrong PR [...]

Argh! You're right! Stupid me 🤦  Let me try again ... #2172! Of course #2172! 😁 

#2172 contributed the code, is merged and contained the "Fixes" line.
